src/App.jsx

```diff
@@ -1,14 +1,38 @@
+import { useState } from 'react';
 import './App.css';
+import ChatLog from './components/ChatLog';
+import DATA from './data/messages.json';
 
+// const
 const App = () => {
+  const [chatData, setChatData] = useState(DATA);
+
+  const handleLikeClick = (id) => {
+    setChatData((prevChatData) => {
+      const updatedChat = prevChatData.map((chat) => {
+        if (chat.id === id) {
+          const isNowLiked = !chat.liked; // Toggle the like status
+          return { ...chat, liked: isNowLiked };
+        }
+        return chat;
+      });
+      return updatedChat;
```

Comment:
We actually don't need to return `updatedChat`. Also, we don't need to use the updater function pattern here for setting the state because this updated state does not need to rely on the previous state. 

Therefore, you could refactor `handleLikeClick` to look like: 
```suggestion
    setChatData((chatData) => 
      chatData.map((chat) => {
        if (chat.id === id) {
          const isNowLiked = !chat.liked; // Toggle the like status
          return { ...chat, liked: isNowLiked };
        }
        return chat;
      })
```